### PR TITLE
refactor(formats): remove unnecessary schema argument from schema inference

### PR DESCRIPTION
**Pull request by kszucs**
`sch.infer()` and `DataMapper.infer_table()` implementations had an unnecessary schema argument which wan't actually used.

### CODE REVIEW

**Review by cpcloud — APPROVED** [review]

File: ibis/formats/__init__.py [hunk 1]
```diff
@@ -168,15 +168,13 @@ def convert_column(cls, obj: C, dtype: DataType) -> C:
         raise NotImplementedError
 
     @classmethod
-    def convert_table(cls, obj: T, schema: Schema) -> T:
+    def convert_table(cls, obj: T) -> T:
```
Comment (gforsyth) [inline ibis/formats/__init__.py:171]: This now seems like a nonsensical function.  Convert it how?  and to what?  

If every child class that uses this defines a schema argument, shouldn't we keep it here?

Comment (cpcloud) [reply ibis/formats/__init__.py:171]: Good catch, seems like this should stay.

Comment (gforsyth) [reply ibis/formats/__init__.py:171]: I can put up a PR to restore it

Comment (gforsyth) [reply ibis/formats/__init__.py:171]: https://github.com/ibis-project/ibis/pull/8821

**Pull request merged**